.github/workflows/push-docker-image.yml

```diff
@@ -0,0 +1,40 @@
+name: Push Docker image
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [main]
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  push-image:
+    name: Push image
+    runs-on: ubuntu-latest
+    environment: ${{ matrix.environment }}
+    if: ${{ github.actor != 'dependabot[bot]' }}
```

Comment:
I've found that this check means that dependabot PRs can't report a success for the required check push-image and therefore can't be merged ([eg in nino smoke tests](https://github.com/govuk-one-login/ipv-cri-check-hmrc-smoke-tests/pull/86))

I've updated the workflow like [this](https://github.com/govuk-one-login/ipv-cri-check-hmrc-smoke-tests/pull/87) which hasn't been reviewed yet but would be keen to get your thoughts and maybe implement a similar thing here

Reply:
This looks good - we're adding it to this PR 👍